services/QuillLMS/spec/models/teacher_info_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: teacher_infos
+#
+#  id                  :bigint           not null, primary key
+#  minimum_grade_level :integer
+#  maximum_grade_level :integer
+#  teacher_id          :bigint
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#
+require 'rails_helper'
+
+describe TeacherInfo, type: :model, redis: true do
+  it { should have_many(:subject_areas) }
+  it { should have_many(:teacher_info_subject_areas) }
+  it { should belong_to(:teacher) }
+
+  it {should validate_numericality_of(:minimum_grade_level)}
+  it {should validate_numericality_of(:maximum_grade_level)}
+
+  describe 'minimum_grade_level=' do
```

Comment:
I definitely did a double take on these specs, but I think they communicate well the subtlety of the approach.